moshi/src/main/java/com/squareup/moshi/Moshi.java

```diff
@@ -70,6 +71,19 @@
         Collections.singleton(Types.createJsonQualifierImplementation(annotationType)));
   }
 
+  @CheckReturnValue
+  public <T> JsonAdapter<T> adapter(Type type, Class<? extends Annotation>... annotationTypes) {
+    if (annotationTypes.length == 1) {
+      return adapter(type,
```

Comment:
maybe `return adapter(type, annotationTypes[0])`
It's interesting that we know `annotationTypes.length` is never 0 because of the existing method signature.

Reply:
That's clever, and saves a bit of code duplication - 5ed2a58